### PR TITLE
update nim / nimrod to 0.11.0; fix install; add test

### PR DESCRIPTION
**Pull request by dcreemer**
Update to upstream nim source 0.11.0 from http://nim-lang.org/download.html
switch from .zip file to .tar.xz as the latter is about 10 times smaller
fix the install phase to install the nim binary
fix the tests to actually test correct output

### CODE REVIEW

**Comment by tdsmith** [comment]
ping @philip-wernersbach per #39217 (issue titled "Please do not version bump the nimrod formula.")

**Comment by dcreemer** [comment]
I will of course defer to the pkg maintainer here -- my goal is just to run the [latest release](https://github.com/Araq/Nim/tree/v0.11.0)  of https://github.com/Araq/Nim on Mac OS X. Does this pull-request make the situation of not having a strategy on https://github.com/nim-lang/nimble sorted out any worse than it is with the [previous release](https://github.com/Araq/Nim/tree/v0.10.2)?

**Comment by tdsmith** [comment]
I'm tempted to pull this until we get a PR that adds nimble, yes.

**Comment by philip-wernersbach** [comment]
Some comments:

1. Remove Lines 7-12, brewbot will have to generate new bottles.
2. If Line 18 is needed to install the `bin` directory, then that's a pretty severe upstream regression, as `install.sh` is supposed to do it. Keep the line and file an upstream issue for it.
3. Remove Line 19, the bug it works around has been fixed upstream.
4. Looks good otherwise.
5. Love the way you've implemented an actual test.

As for #39217, I haven't heard anything from upstream yet. I do have a solution for the upstream blocker in the form of [`nimble-wrapper`](https://github.com/philip-wernersbach/nimble-wrapper), but it was [rejected last time](https://github.com/Homebrew/homebrew/pull/34693) because it would be a Homebrew only fix, which I kind of agree with.

@dcreemer and @tdsmith I'm starting to agree with you two about pulling this for now and maybe adding nimble later when the upstream stuff is fixed. It will create issues for upstream support because they'll expect people to have `nimble` installed by default, but I'm starting to agree that it's better to provide the newest version.

**Comment by tdsmith** [comment]
Please don't modify the bottle block in a pull request, thanks.

Tim

**Comment by dcreemer** [comment]
Thanks for the quick response. I've filed a bug -- Araq/Nim#2647 for the install.sh script. I don't think I can remove line 19 (yet) as this formula refers directly to the 0.11.0 release (and it's tarball) which still have the bug. I'll be happy to update when 0.11.1 (or whatever) is ready.

**Comment by philip-wernersbach** [comment]
Line 19 works around Araq/Nim#1706.

**Comment by def-** [comment]
This has been fixed in 0.11.2.

**Comment by dcreemer** [comment]
I've updated to 0.11.2 as you can see in the recent commit -- this fixes the 'nim' binary install problem. However the 'compiler' directory is still not installing as part of the default script, so I have left that line in place. (Hopefully this is temporary).

**Comment by dcreemer** [comment]
Ah -- I more carefully read the Araq/Nim#1706 bug and see that the compiler directory no longer needs to be installed. Removed.

**Comment by dcreemer** [comment]
ping -- can this be merged?

File: Library/Formula/nimrod.rb [hunk 1]
```diff
@@ -15,16 +15,15 @@ def install
...
   end
 
   test do
     (testpath/"hello.nim").write <<-EOS.undent
-      echo("Hi!")
+      echo("hello")
     EOS
-    system "#{bin}/nim", "compile", "--run", "hello.nim"
+    assert_equal "hello\n", `#{bin}/nim compile --verbosity:0 --run #{testpath}/hello.nim`
+    assert_equal 0, $?.exitstatus
```
Comment (xu-cheng) [inline Library/Formula/nimrod.rb]: You can replace these two line with:
```ruby
assert_equal "hello", shell_output("#{bin}/nim compile --verbosity:0 --run #{testpath}/hello.nim").chomp
```

**Comment by philip-wernersbach** [comment]
Yes, merge this.

Also, no word from upstream yet.

**Comment by dcreemer** [comment]
thanks @xu-cheng -- I have incorporated your suggestion.

**Pull request closed without merging**